dnscache.py

```diff
@@ -163,22 +162,18 @@ def get(self, host, port=0):
                     continue
 
                 ip = records[-1][0]
-                if not ip:
```

Comment:
this conditional is crucial for the deletion, running the removal without it would mean every record gets deleted from the cache, no matter if it contains a valid set of data or not.